src/shared/directives/conditional-show/conditional-show-form.directive.ts

```diff
@@ -0,0 +1,158 @@
+ *  The show condition is evaluated on all the fields of the page. i.e. read only and form fields.
+ *  Evaluation of the show condition includes disabled fields, which can be on their initial value or empty. Executes on the
+ *  host field initialization and when any field of the form changes.
+ *  Collaborates with the GreyBarService to show a vertical grey bar when a field initially hidden on the page is shown. When returning
+ *  to the page after the page has been left, the grey bar has to be redisplayed. If instead on initial page load the field renders as
+ *  initially shown, the grey bar is not displayed.
+ */
+export class ConditionalShowFormDirective implements OnInit, OnDestroy {
+
+  @Input() caseFields: CaseField[];
+  @Input() contextFields: CaseField[] = [];
+  @Input() formGroup: FormGroup;
+
+  private allFieldValues: any;
+  private formChangesSubscription: Subscription;
+
+  constructor(private el: ElementRef,
+              private fieldsUtils: FieldsUtils,
+              private registry: ConditionalShowRegistrarService,
+              private renderer: Renderer2,
+              private greyBarService: GreyBarService) {
+  }
+
+  ngOnInit() {
+    if (!this.formGroup) {
+      console.log ('**** no form group in conditional-show-form directive');
+      this.formGroup = new FormGroup({});
+    }
+    this.evalAllShowHideConditions();
+    this.subscribeToFormChanges();
+  }
+
+  ngOnDestroy() {
+    this.unsubscribeFromFormChanges();
+  }
+
+  private subscribeToFormChanges() {
+    this.unsubscribeFromFormChanges();
+    this.formChangesSubscription = this.formGroup.valueChanges.subscribe(_ => {
+      this.evalAllShowHideConditions();
+    });
+  }
+
+  private evaluateControl(control: AbstractControl) {
+    const cf = control['caseField'] as CaseField;
+    const component = control['component'] as AbstractFormFieldComponent;
+    this.evaluateCondition(cf, component, control);
+  }
+
+  private evaluateCondition(cf: CaseField, component: AbstractFormFieldComponent, control: AbstractControl) {
+    if (cf) {
+      if (cf.show_condition) {
+        const condResult = ShowCondition.getInstance(cf.show_condition).match(this.allFieldValues, this.buildPath(component, cf));
+        if (cf.hidden === null || cf.hidden === undefined) {
+          cf.hidden = false;
+        }
+        if (condResult === cf.hidden) {
+          if (cf.hidden) {
+            this.greyBarService.addToggledToShow(cf.id);
+          } else {
+            this.greyBarService.removeToggledToShow(cf.id)
+          }
+          cf.hidden = !condResult;
+        }
+        // Disable the control if it's hidden so that it doesn't count towards the
+        // validation state of the form, but only if it's actually being validated.
+        if (control.validator) {
+          if (cf.hidden === true && !control.disabled) {
+            control.disable();
+          } else if (cf.hidden !== true && control.disabled) {
+            control.enable();
```

Comment:
Consider using `control.enable({emitEvent: false})` unless event emission is required

Reply:
This has been done and should be in the codebase already.